src/Rule/Ip.php

```diff
@@ -25,11 +24,23 @@
  * ['ip_address', 'ip', 'expandIPv6' => true], // expands IPv6 address to a full notation format
  * ```
  *
- * @property array $ranges The IPv4 or IPv6 ranges that are allowed or forbidden. See [[setRanges()]] for
+ * @property array $ranges The IPv4 or IPv6 ranges that are allowed or forbidden. See [[ranges()]] for
  * detailed description.
  */
 class Ip extends Rule
 {
+    const IPV4 = 4;
```

Comment:
Visibility for constants should be specified.

Reply:
The Ip validator has since been fixed for yii3. PR https://github.com/yiisoft/validator/pull/32